test/integration/integration_test.cc

```diff
@@ -193,31 +193,35 @@ TEST_P(IntegrationTest, WebSocketConnectionDownstreamDisconnect) {
   IntegrationTcpClientPtr tcp_client;
   FakeRawConnectionPtr fake_upstream_connection;
   const std::string upgrade_req_str = "GET /websocket/test HTTP/1.1\r\nHost: host\r\nConnection: "
-                                      "Upgrade\r\nUpgrade: websocket\r\n\r\n";
+                                      "keep-alive, Upgrade\r\nUpgrade: websocket\r\n\r\n";
   const std::string upgrade_resp_str =
       "HTTP/1.1 101 Switching Protocols\r\nConnection: Upgrade\r\nUpgrade: websocket\r\n\r\n";
 
   tcp_client = makeTcpConnection(lookupPort("http"));
   // Send websocket upgrade request
   // The request path gets rewritten from /websocket/test to /websocket.
-  // The size of headers received by the destination is 225 bytes.
+  // The size of headers received by the destination is 228 bytes.
   tcp_client->write(upgrade_req_str);
   fake_upstream_connection = fake_upstreams_[0]->waitForRawConnection();
-  fake_upstream_connection->waitForData(225);
+  const std::string data = fake_upstream_connection->waitForData(228);
+  // The chunked transfer-encoding for no body upstream request must valid
+  if (std::string::npos != data.find("transfer-encoding: chunked")) {
+    EXPECT_NE(std::string::npos, data.find("\r\n\r\n0\r\n"));
```

Comment:
It's not really clear what this is testing for, can you search for the entire "content-length" etc. header? Same below.

Reply:
Done